File: Greedy/PerimeterWithCondition/perimeterWithCondition.js
(function () {


    (function() {

        let arr = [1,1,1,3,3];
        arr.sort(function(a,b) {
            if(a < b) return -1;
        });

        let start = arr.length - 3;
        let mid = arr.length - 2;
        let end = arr.length - 1;
        while(start+mid <= end) {
            start--;
            mid--;
            end--;
            if(start < 0) {
                console.log("-1");
                return;
            }
        }
        console.log(arr[start], " ", arr[mid], " ", arr[end]);

    }());    
}());

File: Greedy/PerimeterWithCondition/perimeterWithCondition.cpp
#include<iostream>
#include<algorithm>
using std::cin;
using std::cout;
using std::endl;
using std::sort;

int main() {
    int n;
    cin >> n;
    int* arr{new int[n]};
    for(int i{}; i < n; i++) cin >> arr[i];
    sort(arr, arr + n);
    int start{n - 3}, mid{n - 2}, end{n - 1};
    while(start + mid <= end) {
        start--;
        mid--;
        end--;
        if(start < 0) {
            cout << "-1" << endl;
            return 0;
        }
    }
    cout << arr[start] << " " << arr[mid] << " " << arr[end] << endl;
}
